network/src/interface/mod.rs

```diff
@@ -379,21 +378,22 @@ where
 
     async fn handle_peer_mgr_notification(
         notif: PeerManagerNotification<TSubstream>,
-        mut upstream_handlers: HashMap<ProtocolId, channel::Sender<NetworkNotification>>,
+        mut upstream_handlers: HashMap<
+            ProtocolId,
+            libra_channel::Sender<PeerId, NetworkNotification>,
+        >,
     ) {
         trace!("PeerManagerNotification::{:?}", notif);
         match notif {
             PeerManagerNotification::NewPeer(peer_id, _addr) => {
                 for ch in upstream_handlers.values_mut() {
-                    ch.send(NetworkNotification::NewPeer(peer_id))
-                        .await
```

Comment:
this is what makes me very nervous about this refactor, since health_checker and discovery both require reliable delivery of these control messages. maybe something we need to do with the QoS refactor is split out control messages from application messages.